packages/adapters-library/src/adapters/prices-v2/products/usd/pricesV2UsdAdapter.ts

```diff
@@ -108,16 +105,24 @@ export class PricesV2UsdAdapter implements IPricesAdapter {
   private async getSpotPriceByAddress({
     tokenAddresses,
     vsCurrency = 'usd',
-    chainId,
   }: GetSpotPriceByAddressInput): Promise<SpotPrice | null> {
-    const baseUrl = 'https://price-api.metafi.codefi.network'
-    const url = `${baseUrl}/v1/chains/${chainId}/spot-prices/${tokenAddresses}`
+    const url = `${BASE_URL}/v1/chains/${this.chainId}/spot-prices/${tokenAddresses}`
 
-    try {
-      const params = new URLSearchParams()
-      if (vsCurrency) params.append('vsCurrency', vsCurrency)
+    const urlObject = new URL(url)
+    urlObject.searchParams.append('vsCurrency', vsCurrency)
+
+    logger.info(
+      {
+        tokenAddresses,
+        currency: vsCurrency,
+        chainId: this.chainId,
+        url: urlObject.toString(),
+      },
+      'Fetching spot prices',
+    )
 
-      const response = await fetch(`${url}?${params.toString()}`, {
```

Comment:
It's safer to build the URL using the URL class (this would fail if querystring params were added to the base url)